packages/opencensus-core/src/tags/tag-map.ts

```diff
@@ -31,6 +31,14 @@ export class TagMap {
     if (!isValidTagValue(tagValue)) {
       throw new Error(`Invalid TagValue: ${tagValue.value}`);
     }
+    let existingKey;
+    for (const key of this.registeredTags.keys()) {
+      if (key.name === tagKey.name) {
+        existingKey = key;
+        break;
+      }
+    }
+    if (existingKey) this.registeredTags.delete(existingKey);
```

Comment:
Just curious, in Node if I want to update the value for a key in a map, I need to delete the original <key, value> first?

Reply:
As per my understanding, this is true if we use ```object``` as a key for ```map```. The maps detect existence ```( Map.prototype.has ( key ))``` of keys based on equality (===). Primitives like strings and numbers are compared by their value, while objects (arrays, plain objects) are compared by their reference. That comparison by reference basically checks to see if the objects given refer to the same location in memory. 

Basically to check the objects' "**value equality**" we essentially have to iterate over every property in the objects to see whether they are equal.

A potential option is to use ```tagkey.name``` as a key which is a string type. This required lot more changes and would like to handle in separate PR if we really want to do it.

Reply:
Make sense, thanks for the explanation. I'd prefer to keep using the `TagKey` object as the map key.